tools/cellprofiler/cellprofiler.xml

```diff
@@ -0,0 +1,75 @@
+<tool id="cp319" name="Cellprofiler" version="19.05">
+  <description>Run a CellProfiler v3.1.9 pipeline</description>
+  <requirements>
+     <requirement type="package" version="3.1.9">cellprofiler</requirement>
+  </requirements>
+  <configfiles>
+<configfile name="script_file">
+mkdir `pwd`/input
+mkdir `pwd`/output
+#for $i,$filename in enumerate($inputType.inputdir):
+ln -s $filename `pwd`/input/${filename.element_identifier}.${filename.ext}
+#end for
+
+find `pwd`/input -name *.tiff > list.txt
+
+</configfile>
+  </configfiles>
+
+  <command><![CDATA[
+      bash '$script_file' &&
+      cellprofiler -c -r --file-list list.txt -o `pwd`/output -p $pipeline 
+  ]]></command>
+  <inputs>
+	  <param name="pipeline" type="data" format="txt" label="Pipeline file"/>
+    
+    <conditional name="inputType">
+       <param name="iType" type="select" label="Select either a zip file or a collection from history">
+          <option value="c">collection</option>
+          <option value="z">zip file</option>
+          <option value="s">images</option>
+       </param>
+       <when value="z">
+          <param name="inputdir" type="data" format="zip" label="Select image dataset from history" /> 
+       </when>
+        <when value="s">
+          <param name="inputdir" type="data" multiple="true" format="tiff,tif" label="Images"/>
+       </when>         
+        <when value="c">
+          <param name="inputdir" type="data_collection" collection_type="list" label="Images"/>
+       </when>       
+    </conditional>
+          
+  </inputs>
+  <outputs>
+   <!--  <data format="txt" name="out_file" />-->
+
+      <collection name="pipeline_output" type="list" label="Cellprofiler pipeline output files">
+        <discover_datasets pattern="__designation_and_ext__" visible="false" directory="output"/>
+      </collection>     
+
+  </outputs>
+
+  <stdio>
```

Comment:
this should be on-top, there is a preferred ordering, `planemo lint` will tell you

Reply:
changed